### ISSUE TITLE
Camera crash when open from real device

### ISSUE THREAD

**Issue by Saavedra22** [flutter/flutter#79293]
**Hi, I am trying to open my camera from a real device and before opening my app it closes. In my emulator everything works fine**
### Code sample
final _picker = ImagePicker();
  void _pickImage() async {
    PickedFile pickedImage = await _picker.getImage(
      source: ImageSource.camera,
      imageQuality: 50,
      maxWidth: 150
    );
    setState(() {
      _pickedImage = File(pickedImage.path);
    });
    widget.imagePickEvent(File(pickedImage.path));
  }
### Console return
Syncing files to device Redmi Note 7...
Reloaded 5 of 1058 libraries in 879ms.
I/.example.prueb(13651): Waiting for a blocking GC ProfileSaver
I/.example.prueb(13651): WaitForGcToComplete blocked ProfileSaver on ProfileSaver for 11.661ms
I/Timeline(13651): Timeline: Activity_launch_request time:463546564
W/ActivityThread(13651): SCHED: com.example.prueba/.MainActivity [84, r=75ms, a=14ms, w=245668ms]
E/AndroidRuntime(16190): FATAL EXCEPTION: main
E/AndroidRuntime(16190): Process: com.google.android.GoogleCamera, PID: 16190
E/AndroidRuntime(16190): java.lang.NullPointerException: Parameter keys cannot be null.
E/AndroidRuntime(16190): 	at com.google.common.collect.Platform.checkNotNull(Platform.java:910)
E/AndroidRuntime(16190): 	at com.google.android.apps.camera.one.core.Request$Parameter.<init>(Request.java:33)
E/AndroidRuntime(16190): 	at com.google.android.apps.camera.one.core.RequestTransformers.forParameter(RequestTransformers.java:64)
E/AndroidRuntime(16190): 	at com.google.android.apps.camera.one.photo.commands.AutoFlashHardwareNoRawZslHdrPlusImageCaptureCommandFactory$HardwareFlashImageCaptureCommand.<init>(AutoFlashHardwareNoRawZslHdrPlusImageCaptureCommandFactory.java:79)
E/AndroidRuntime(16190): 	at com.google.android.apps.camera.legacy.app.one.v2.photo.PictureTakerModules_HardwareNoRawZslHdrPlus_ProvideStateTrackedImageCaptureCommandFactory.get(PictureTakerModules_HardwareNoRawZslHdrPlus_ProvideStateTrackedImageCaptureCommandFactory.java:53)
E/AndroidRuntime(16190): 	at dagger.internal.DoubleCheck.get(DoubleCheck.java:51)
E/AndroidRuntime(16190): 	at com.google.android.apps.camera.one.photo.commands.StateTrackingImageCaptureCommand_Factory.get(StateTrackingImageCaptureCommand_Factory.java:32)
E/AndroidRuntime(16190): 	at dagger.internal.DoubleCheck.get(DoubleCheck.java:51)
E/AndroidRuntime(16190): 	at com.google.android.apps.camera.one.photo.common.PictureTakerModule_ProvideInitializedRootCommandFactory.get(PictureTakerModule_ProvideInitializedRootCommandFactory.java:41)
E/AndroidRuntime(16190): 	at com.google.android.apps.camera.one.photo.common.PictureTakerModule$$Lambda$1.run(Unknown Source:12)
E/AndroidRuntime(16190): 	at com.google.android.apps.camera.async.AsyncTasks$$Lambda$4.run(Unknown Source:43)
E/AndroidRuntime(16190): 	at com.google.android.libraries.camera.async.MainThreadThrowingExecutor$MainThreadThrowingRunnable.run(MainThreadThrowingExecutor.java:21)
E/AndroidRuntime(16190): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
E/AndroidRuntime(16190): 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
E/AndroidRuntime(16190): 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
E/AndroidRuntime(16190): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
E/AndroidRuntime(16190): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
E/AndroidRuntime(16190): 	at java.lang.Thread.run(Thread.java:919)
E/AndroidRuntime(16190): 	at com.google.android.libraries.camera.async.AndroidPriorityThread.run(AndroidPriorityThread.java:33)
W/Activity(13651): Slow Operation: Activity com.example.prueba/.MainActivity onDestroy took 318ms
W/Looper  (13651): Slow Looper main: doFrame is 371ms late because of 1 msg, msg 1 took 392ms (seq=303 running=30ms runnable=6ms io=39ms h=android.app.ActivityThread$H w=159)
D/FlutterLocationService(13651): Unbinding from location service.
D/FlutterLocationService(13651): Destroying service.

### Pubspec.yaml
dependencies:
  flutter:
    sdk: flutter
  intl: ^0.17.0
  provider: ^5.0.0
  http: ^0.13.1
  shared_preferences: ^2.0.5
  image_picker: ^0.7.3
  path_provider: ^2.0.1
  path: ^1.8.0
  sqflite: ^2.0.0+3
  location: ^4.1.1
  google_maps_flutter: ^2.0.1
  cloud_firestore: ^1.0.3
  firebase_core: ^1.0.2
  firebase_auth: ^1.0.1
  firebase_storage: ^8.0.1

**Comment by iapicca** [comment]
@Saavedra22 
if you are experiencing the issue in release mode it might be a duplicate of https://github.com/flutter/flutter/issues/78625
if it occurs in `debug`  as well you may want to try running the [sample app](https://github.com/flutter/plugins/blob/master/packages/camera/camera/example/lib/main.dart) 
making sure your flutter version is the`latest stable` or above

**Comment by Saavedra22** [comment]
Hi, @iapicca, my error only shows up in debug mode and only in my real device, the version I have of flutter is the most recent, since I recently started developing in flutter.

**Comment by iapicca** [comment]
@Saavedra22 
the issue might depends from your implementation rather than flutter,
you may try these steps:
-  clone plugins `https://github.com/flutter/plugins.git`
- `cd plugins/packages/camera/camera/example/`
- `flutter run -v`

<details>
<summary>logs</summary>

```bash
[ +998 ms] ✓  Built build/app/outputs/flutter-apk/app-debug.apk.
[   +3 ms] executing: /home/francesco/Android/Sdk/build-tools/30.0.2/aapt dump xmltree /home/francesco/projects/plugins/packages/camera/camera/example/build/app/outputs/flutter-apk/app.apk
AndroidManifest.xml
[  +23 ms] Exit code 0 from: /home/francesco/Android/Sdk/build-tools/30.0.2/aapt dump xmltree /home/francesco/projects/plugins/packages/camera/camera/example/build/app/outputs/flutter-apk/app.apk
AndroidManifest.xml
[        ] N: android=http://schemas.android.com/apk/res/android
             E: manifest (line=2)
               A: android:versionCode(0x0101021b)=(type 0x10)0x1
               A: android:versionName(0x0101021c)="1.0" (Raw: "1.0")
               A: android:compileSdkVersion(0x01010572)=(type 0x10)0x1d
               A: android:compileSdkVersionCodename(0x01010573)="10" (Raw: "10")
               A: package="io.flutter.plugins.cameraexample" (Raw: "io.flutter.plugins.cameraexample")
               A: platformBuildVersionCode=(type 0x10)0x1d
               A: platformBuildVersionName=(type 0x10)0xa
               E: uses-sdk (line=7)
                 A: android:minSdkVersion(0x0101020c)=(type 0x10)0x15
                 A: android:targetSdkVersion(0x01010270)=(type 0x10)0x1c
               E: uses-feature (line=11)
                 A: android:name(0x01010003)="android.hardware.camera" (Raw: "android.hardware.camera")
                 A: android:required(0x0101028e)=(type 0x12)0xffffffff
               E: uses-permission (line=15)
                 A: android:name(0x01010003)="android.permission.INTERNET" (Raw: "android.permission.INTERNET")
               E: uses-permission (line=16)
                 A: android:name(0x01010003)="android.permission.FLASHLIGHT" (Raw: "android.permission.FLASHLIGHT")
               E: uses-permission (line=17)
                 A: android:name(0x01010003)="android.permission.CAMERA" (Raw: "android.permission.CAMERA")
               E: uses-permission (line=18)
                 A: android:name(0x01010003)="android.permission.RECORD_AUDIO" (Raw: "android.permission.RECORD_AUDIO")
               E: uses-permission (line=19)
                 A: android:name(0x01010003)="android.permission.ACCESS_NETWORK_STATE" (Raw: "android.permission.ACCESS_NETWORK_STATE")
               E: application (line=21)
                 A: android:label(0x01010001)="camera_example" (Raw: "camera_example")
                 A: android:icon(0x01010002)=@0x7f080000
                 A: android:name(0x01010003)="io.flutter.app.FlutterApplication" (Raw: "io.flutter.app.FlutterApplication")
                 A: android:debuggable(0x0101000f)=(type 0x12)0xffffffff
                 A: android:appComponentFactory(0x0101057a)="androidx.core.app.CoreComponentFactory" (Raw: "androidx.core.app.CoreComponentFactory")
                 E: activity (line=27)
                   A: android:theme(0x01010000)=@0x7f0a0000
                   A: android:name(0x01010003)="io.flutter.plugins.cameraexample.EmbeddingV1Activity" (Raw: "io.flutter.plugins.cameraexample.EmbeddingV1Activity")
                   A: android:exported(0x01010010)=(type 0x12)0xffffffff
                   A: android:launchMode(0x0101001d)=(type 0x10)0x1
                   A: android:configChanges(0x0101001f)=(type 0x11)0x24b4
                   A: android:windowSoftInputMode(0x0101022b)=(type 0x11)0x10
                   A: android:hardwareAccelerated(0x010102d3)=(type 0x12)0xffffffff
                   E: meta-data (line=35)
                     A: android:name(0x01010003)="io.flutter.app.android.SplashScreenUntilFirstFrame" (Raw: "io.flutter.app.android.SplashScreenUntilFirstFrame")
                     A: android:value(0x01010024)=(type 0x12)0xffffffff
                 E: activity (line=39)
                   A: android:theme(0x01010000)=@0x7f0a0000
                   A: android:name(0x01010003)="io.flutter.embedding.android.FlutterActivity" (Raw: "io.flutter.embedding.android.FlutterActivity")
                   A: android:launchMode(0x0101001d)=(type 0x10)0x1
                   A: android:configChanges(0x0101001f)=(type 0x11)0x24b4
                   A: android:windowSoftInputMode(0x0101022b)=(type 0x11)0x10
                   A: android:hardwareAccelerated(0x010102d3)=(type 0x12)0xffffffff
                   E: intent-filter (line=46)
                     E: action (line=47)
                       A: android:name(0x01010003)="android.intent.action.MAIN" (Raw: "android.intent.action.MAIN")
                     E: category (line=49)
                       A: android:name(0x01010003)="android.intent.category.LAUNCHER" (Raw: "android.intent.category.LAUNCHER")
                 E: meta-data (line=53)
                   A: android:name(0x01010003)="flutterEmbedding" (Raw: "flutterEmbedding")
                   A: android:value(0x01010024)=(type 0x10)0x2
[  +17 ms] Stopping app 'app.apk' on Pixel 3a.
[        ] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C shell am force-stop io.flutter.plugins.cameraexample
[ +127 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C shell pm list packages io.flutter.plugins.cameraexample
[ +112 ms] Installing APK.
[   +3 ms] Installing build/app/outputs/flutter-apk/app.apk...
[        ] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C install -t -r /home/francesco/projects/plugins/packages/camera/camera/example/build/app/outputs/flutter-apk/app.apk
[+4900 ms] Performing Streamed Install
                    Success
[        ] Installing build/app/outputs/flutter-apk/app.apk... (completed in 4.9s)
[   +1 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C shell echo -n 9be278f8dc9ba7d9c8a3a09216c0e5bc45467604 > /data/local/tmp/sky.io.flutter.plugins.cameraexample.sha1
[  +31 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C shell -x logcat -v time -t 1
[ +194 ms] --------- beginning of system
                    03-29 22:12:05.079 I/ActivityManager( 1619): Start proc 21899:com.google.android.packageinstaller/u0a59 for broadcast
                    {com.google.android.packageinstaller/com.android.packageinstaller.PackageInstalledReceiver}
[  +28 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C shell am start -a android.intent.action.RUN -f 0x20000000 --ez enable-background-compilation true --ez
enable-dart-profiling true --ez enable-checked-mode true --ez verify-entry-points true io.flutter.plugins.cameraexample/io.flutter.embedding.android.FlutterActivity
[  +99 ms] Starting: Intent { act=android.intent.action.RUN flg=0x20000000 cmp=io.flutter.plugins.cameraexample/io.flutter.embedding.android.FlutterActivity (has extras) }
[        ] Waiting for observatory port to be available...
[+2063 ms] Observatory URL on device: http://127.0.0.1:37477/CsFu0n6egws=/
[   +1 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward tcp:0 tcp:37477
[   +8 ms] 43283
[        ] Forwarded host port 43283 to device port 37477 for Observatory
[   +9 ms] Caching compiled dill
[ +110 ms] Connecting to service protocol: http://127.0.0.1:43283/CsFu0n6egws=/
[ +743 ms] Launching a Dart Developer Service (DDS) instance at http://127.0.0.1:0, connecting to VM service at http://127.0.0.1:43283/CsFu0n6egws=/.
[ +485 ms] DDS is listening at http://127.0.0.1:37061/JHFFDOFVQcs=/.
[ +107 ms] Successfully connected to service protocol: http://127.0.0.1:43283/CsFu0n6egws=/
[  +48 ms] DevFS: Creating new filesystem on the device (null)
[  +90 ms] I/CameraManagerGlobal(21952): Connecting to camera service
[  +52 ms] DevFS: Created new filesystem on the device (file:///data/user/0/io.flutter.plugins.cameraexample/code_cache/exampleUACPMW/example/)
[   +6 ms] Updating assets
[ +116 ms] Syncing files to device Pixel 3a...
[   +1 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[   +2 ms] <- recompile package:camera_example/main.dart 0f298141-5e6d-4f63-a64b-8250c370dbec
[        ] <- 0f298141-5e6d-4f63-a64b-8250c370dbec
[ +110 ms] Updating files.
[        ] DevFS: Sync finished
[   +1 ms] Syncing files to device Pixel 3a... (completed in 115ms)
[        ] Synced 0.0MB.
[        ] <- accept
[  +21 ms] Connected to _flutterView/0x77f3fc17b0.
[   +2 ms] Flutter run key commands.
[   +1 ms] r Hot reload. 🔥🔥🔥
[   +1 ms] R Hot restart.
[        ] h Repeat this help message.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] 💪 Running with sound null safety 💪
[        ] An Observatory debugger and profiler on Pixel 3a is available at: http://127.0.0.1:37061/JHFFDOFVQcs=/
[+1707 ms] The Flutter DevTools debugger and profiler on Pixel 3a is available at: http://127.0.0.1:9100?uri=http%3A%2F%2F127.0.0.1%3A37061%2FJHFFDOFVQcs%3D%2F
[+1556718 ms] I/Gralloc4(21952): mapper 4.x is not supported
[        ] W/Gralloc3(21952): mapper 3.x is not supported
[+7284 ms] I/Camera  (21952): [FPS Range Available] is:[15, 15]
[        ] I/Camera  (21952): [FPS Range Available] is:[7, 30]
[        ] I/Camera  (21952): [FPS Range Available] is:[15, 30]
[        ] I/Camera  (21952): [FPS Range Available] is:[30, 30]
[        ] I/Camera  (21952): [FPS Range Available] is:[15, 60]
[        ] I/Camera  (21952): [FPS Range Available] is:[60, 60]
[        ] I/Camera  (21952): [FPS Range] is:[15, 60]
[ +856 ms] W/Gralloc4(21952): allocator 3.x is not supported
[        ] W/Gralloc3(21952): allocator 3.x is not supported
[+3447 ms] I/Camera  (21952): [FPS Range Available] is:[15, 15]
[        ] I/Camera  (21952): [FPS Range Available] is:[7, 30]
[        ] I/Camera  (21952): [FPS Range Available] is:[15, 30]
[        ] I/Camera  (21952): [FPS Range Available] is:[30, 30]
[        ] I/Camera  (21952): [FPS Range] is:[7, 30]
[+2922 ms] I/Camera  (21952): [FPS Range Available] is:[15, 15]
[        ] I/Camera  (21952): [FPS Range Available] is:[7, 30]
[        ] I/Camera  (21952): [FPS Range Available] is:[15, 30]
[        ] I/Camera  (21952): [FPS Range Available] is:[30, 30]
[        ] I/Camera  (21952): [FPS Range Available] is:[15, 60]
[        ] I/Camera  (21952): [FPS Range Available] is:[60, 60]
[        ] I/Camera  (21952): [FPS Range] is:[15, 60]
[+18898 ms] DevFS: Deleting filesystem on the device (file:///data/user/0/io.flutter.plugins.cameraexample/code_cache/exampleUACPMW/example/)
[  +24 ms] DevFS: Deleted filesystem on the device (file:///data/user/0/io.flutter.plugins.cameraexample/code_cache/exampleUACPMW/example/)
[ +283 ms] Service protocol connection closed.
[        ] Application finished.
[   +3 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --list
[   +9 ms] Exit code 0 from: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --list
[        ] 965AY0WP5C tcp:43283 tcp:37477
[   +1 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --remove tcp:43283
[  +10 ms] executing: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --list
[   +7 ms] Exit code 0 from: /home/francesco/Android/Sdk/platform-tools/adb -s 965AY0WP5C forward --list
[   +2 ms] "flutter run" took 1,714,488ms.
[ +255 ms] ensureAnalyticsSent: 251ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 0
```

</details>

<details>
<summary>doctor</summary>

```console
[✓] Flutter (Channel master, 2.1.0-13.0.pre.294, on Linux, locale en_US.UTF-8)
    • Flutter version 2.1.0-13.0.pre.294 at /home/francesco/snap/flutter/common/flutter
    • Framework revision a603714610 (33 hours ago), 2021-03-28 03:54:02 -0700
    • Engine revision b5e15d055d
    • Dart version 2.13.0 (build 2.13.0-162.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/francesco/Android/Sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_SDK_ROOT = /home/francesco/Android/Sdk
    • Java binary at: /usr/lib/jvm/java-1.8.0-openjdk-amd64/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_282-8u282-b08-0ubuntu1~20.04-b08)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /snap/bin/chromium

[✓] Linux toolchain - develop for Linux desktop
    • clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
    • cmake version 3.10.2
    • ninja version 1.8.2
    • pkg-config version 0.29.1

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[✓] Connected device (3 available)
    • Pixel 3a (mobile) • 965AY0WP5C • android-arm64  • Android 11 (API 30)
    • Linux (desktop)   • linux      • linux-x64      • Linux
    • Chrome (web)      • chrome     • web-javascript • Chromium 89.0.4389.90 snap

! Doctor found issues in 1 category.
```

</details>

the issue doesn't seem to be caused by [camera](https://pub.dev/packages/camera) plugin or flutter itself

**Comment by Saavedra22** [comment]
@iapicca Thank you very much for the contribution, I was reviewing the examples and I realized that at least on my device the **image_picker** plugin does not work correctly, but the **camera* plugin works without any problem. _Thank you_
